### PR TITLE
feat(lsp): add `jump_type` config option for location handler

### PR DESCRIPTION
**Pull request by pmizio**
Hi, it's my first pr!
PR add config option for location handlers eg. `textDocument/definition` to open location in another type of window. If `jump_type` is `nil` handler behaves like it is working now.
```lua
-- for one mapping
vim.keymap.set("n", "gv", function()
  vim.lsp.buf.definition { jump_type = "vsplit" }
end, { buffer = bufnr })

-- or for all of them
vim.lsp.handlers["textDocument/definition"] = vim.lsp.with(
  vim.lsp.handlers.location, {
    -- Opens definition in new split
    jump_type = "vsplit"
  }
)
```

### CODE REVIEW

**Comment by justinmk** [comment]
similar to https://github.com/neovim/neovim/issues/18625#issuecomment-1130504199 , I'm wondering if this is another case where you just need to run a custom callback on various lsp events. 

This PR is adding `vim.cmd()` at a particular place. No real reason that is only for "jumps".

File: runtime/lua/vim/lsp/handlers.lua [hunk 1]
```diff
@@ -321,13 +321,28 @@ end
...
+---     -- Opens definition in new split
+---     jump_type = "vsplit"
+---   }
+--- )
+--- </pre>
+---@param config table Configuration table.
+---     - jump_type:     (default=nil)
+---         - Opens definition in window opened using this command
+---         - See |opening-window|
+function M.location(_, result, ctx, config)
```
Comment (justinmk) [inline runtime/lua/vim/lsp/handlers.lua:345]: why make it public?

**Review by justinmk — APPROVED** [review]
.

**Review by justinmk — CHANGES_REQUESTED** [review]
- check if there are existing ways to "do something after an LSP action"
- check if a general feature like an `LspRequest` / `LspAction` events would address this use-case

**Review by mfussenegger — COMMENTED** [review]
You could accomplish the same already by using the `vim.lsp.tagfunc` and then something like `CTLR-w ]`.

Not sure there is a benefit to this approach here?
If so, I think the approach started in https://github.com/neovim/neovim/pull/18577 should be used, because the global handler will likely become unused by default once those functions are adapted to support multiple clients.

**Pull request closed without merging**